### DISCUSSION TITLE
Authentication trouble (noob questions)

### DISCUSSION THREAD

**Discussion by BionicAbeLincoln** [Murmele/Gittyup#227] [Q&A] [answered]
Hi, I'm new to Git (previous projects have been TFS and Mercurial), and I've been trying new Git clients and this one looks really nice!  I am using a repository hosted on Azure Devops, and when fetching Gittyup asks me for credentials under the username of my organization rather than my user.  After banging my head against it and entering the org name with my password, my name with my password, my email with my pass, etc, I found I could fetch if I went onto the repo web page, clicked the "Clone" button and then "Generate Git Credentials".  This gave me a set of credentials with my username and a big hashed password, and the client accepted that.  I'd hoped I'd established authentication with that, but Gittyup still asks for credentials with every fetch.  On other clients, I've entered my password and username when first pulling maybe once and then it would be remembered.  Am I doing something wrong?

**Comment by exactly-one-kas** [comment, marked as answer]
Which OS are you using?

**Reply by BionicAbeLincoln** [reply]
Windows 10

**Reply by exactly-one-kas** [reply]
I'll try to look into that on monday; on Windows passwords should be stored in and retrieved from the Windows password vault

**Reply by exactly-one-kas** [reply]
On a fresh install it works flawlessly.

Is storing credentials turned on in the settings?
![2022-08-08 19-37-22](https://user-images.githubusercontent.com/16035731/183479296-0bc828da-5aea-4633-baec-848dd6473a4b.png)


If it is, please enable credential logging using the registry key `HKEY_CURRENT_USER\Software\gittyup.github.com\Gittyup\credential\log` by adding it as a string with its value set to `true` (and restarting Gittyup)
![grafik](https://user-images.githubusercontent.com/16035731/183483112-4320578e-72c6-4d55-907f-b4e526362b6e.png)
After that, open `%TEMP%\Gittyup\cred.log`, censor any private information and paste it here

**Reply by BionicAbeLincoln** [reply]
Interesting, store credentials was checked in settings but there was no `credential` key in the registry.  I unchecked store credentials, restarted gittyup, then re-checked it, and now there is a credential key.  I tried fetching again and entered my credentials again, but it still didn't remember them.  So I added the `log` key as instructed and restarted the client and attempted to fetch a few times, but I can't find the `%TEMP%` folder.  Is it supposed to be in C:/ or something?

**Reply by exactly-one-kas** [reply]
`%TEMP%` is an environment variable, not a literal folder name

You can type `%TEMP%` into the Windows Explorer address bar and it should take you to your user's temporary files folder

**Reply by BionicAbeLincoln** [reply]
Gotcha; here's the censored log

> 14:36:21.794 - get: ***Azure DevOps Address*** [OrganizationName]
14:36:21.797 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:36:34.550 - store: ***Azure DevOps Address*** (username)
14:36:44.563 - get: ***Azure DevOps Address*** [OrganizationName]
14:36:44.574 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:36:54.226 - store: ***Azure DevOps Address*** (username)
14:39:51.453 - get: ***Azure DevOps Address*** [OrganizationName]
14:39:51.463 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:39:57.811 - store: ***Azure DevOps Address*** (usernameWithTypoWhoops)
14:41:15.173 - get: ***Azure DevOps Address*** [OrganizationName]
14:41:15.174 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:41:21.025 - store: ***Azure DevOps Address*** (username)
14:41:56.855 - get: ***Azure DevOps Address*** [OrganizationName]
14:41:56.867 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:42:01.096 - store: ***Azure DevOps Address*** (username)
14:42:19.043 - get: ***Azure DevOps Address*** [OrganizationName]
14:42:19.054 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:42:23.743 - store: ***Azure DevOps Address*** (username)
14:49:38.462 - get: ***Azure DevOps Address*** [OrganizationName]
14:49:38.474 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'
14:49:45.371 - store: ***Azure DevOps Address*** (username)
14:49:46.874 - get: ***Azure DevOps Address*** [OrganizationName]
14:49:46.887 - get: credential not found for 'git:https://[OrganizationName]@dev.azure.com'

**Reply by exactly-one-kas** [reply]
Can you try replacing the organization name in the repository's remote address part before the @ with your username so that it starts with `https://username@dev.azure.com`?

**Reply by BionicAbeLincoln** [reply]
I think that may have done it!  Thanks very much for your help

**Reply by exactly-one-kas** [reply]
Great!
Remember to turn off credential logging by simply deleting the `log` registry key
